### PR TITLE
3.next - Enable scoped middleware to be setup

### PR DESCRIPTION
**Pull request by markstory**
Add the methods that help developers register and configure scoped middleware for their applications. The next task will be to use these new features when dispatching requests.

Refs #10308

### CODE REVIEW

File: src/Routing/RouteBuilder.php [hunk 1]
```diff
@@ -714,4 +714,38 @@ public function fallbacks($routeClass = null)
...
+    /**
+     * Apply a middleware to the current route scope.
+     *
+     * Requires middleware to be registered via `registerMiddleware()`
+     *
+     * @param string[] ...$names The names of the middleware to apply to the current scope.
+     * @return $this
+     * @see \Cake\Routing\RouteCollection::addMiddlewareToScope()
+     */
+    public function middleware(...$names)
```
Comment (cleptric) [inline src/Routing/RouteBuilder.php]: Maybe `applyMiddleware()` or `enableMiddleware()` to better display what the method does?

Comment (markstory) [inline src/Routing/RouteBuilder.php]: I like `applyMiddleware()`.

Comment (cleptric) [inline src/Routing/RouteBuilder.php]: So mote it be 😀

**Comment by markstory** [comment]
Merging this so I can do the next step.

**Pull request merged**